src/mmap_storage.cpp

```diff
@@ -904,7 +904,14 @@ error_code translate_error(std::system_error const& err, bool const write)
 
 		boost::optional<aux::file_view> h = open_file_impl(sett, file, mode, ec);
 		if ((mode & aux::open_mode::write)
-			&& ec.ec == boost::system::errc::no_such_file_or_directory)
+			&& (ec.ec == boost::system::errc::no_such_file_or_directory
+#ifdef _WIN32
+				// this is a workaround for improper handling of files on windows shared drives.
+				// if the directory on a shared drive does not exist, 
+				// windows returns ERROR_IO_DEVICE instead of ERROR_FILE_NOT_FOUND
+				|| ec.ec == error_code(ERROR_IO_DEVICE, boost::asio::error::system_category)
```

Comment:
I would prefer if this was unqualified, to pick up `libtorrent::system_category()`, and actually calling the function.  `system_category()`

I'm not sure why this compiles at all.

Reply:
There must be something weird with my compile configuration such as boost version, because this compiles for me but yours didn't.

If it's ok with you why don't you implement it in a way you prefer, because I am not really knowledgeable enough about the code base to do it in a way I'm confident in

Reply:
how does this look? https://github.com/arvidn/libtorrent/pull/6718